closed/src/java.base/share/classes/sun/security/provider/NativeDigest.java

```diff
@@ -70,24 +87,24 @@
         this.digestLength = digestLength;
         this.algIndx = algIndx;
         this.context = nativeCrypto.DigestCreateContext(0, algIndx);
+        digestCleaner.register(this, new DigestCleanerRunnable(this.context));
```

Comment:
The call to `register` should move after the validation check (also in `clone()`).